### PR TITLE
Move out most of "non-fully active" handling guide

### PR DESCRIPTION
**Pull request by rakina**
The guide is being moved to a separate "Supporting BFCached documents" doc, to keep the design principles doc short. Replaces the previous content with a short intro and link to the full guide (which is still in my personal repo). See also https://github.com/w3ctag/design-reviews/issues/628#issuecomment-1227165134 for context.

***
<a href="https://pr-preview.s3.amazonaws.com/rakina/design-principles/pull/392.html" title="Last updated on Dec 6, 2022, 8:48 AM UTC (f314c87)">Preview</a> | <a href="https://pr-preview.s3.amazonaws.com/w3ctag/design-principles/392/a6266b8...rakina:f314c87.html" title="Last updated on Dec 6, 2022, 8:48 AM UTC (f314c87)">Diff</a>

### CODE REVIEW

**Comment by LeaVerou** [comment]
Hi there, I noticed that the link to the new guide is TODO. I think we should not merge this until that is available, otherwise we're just removing useful guidance.

**Comment by rakina** [comment]
Yep, the plan is to land the BFCache Guide doc first before landing the update to these existing docs.

Is there a good way to get the BFCache Guide doc reviewed? Currently it lives in https://github.com/rakina/bfcache-guide and I can't really make a PR for it because it's a new doc... (How did the Promises guide get reviewed?)

**Comment by rakina** [comment]
Friendly ping for the review for this and the [other PRs](https://github.com/w3ctag/design-reviews/issues/628#issuecomment-1227165134). Also still don't know where to put the new BFCache doc for review... (@domenic, how did you get to make the https://github.com/w3ctag/promises-guide repo?)

**Comment by domenic** [comment]
> (@domenic, how did you get to make the https://github.com/w3ctag/promises-guide repo?)

I think I created a separate repo under my own GitHub, and eventually transferred it to the w3ctag. So you're on the right track...

File: index.bs [hunk 1]
```diff
@@ -691,183 +691,23 @@ it is not always *sufficient*
...
+* May malfunction when a document is kept in a non-[=Document/fully active=] BFCached state (instead of getting destroyed)
+    after the user navigates away from it or gets restored (e.g. expects that a state saved in the document won’t span multiple navigations)
 
-<h4 id="omit-non-fully-active">Omit non-[=Document/fully active=] documents from APIs that span multiple documents</h4>
-Non-[=Document/fully active=] documents should not be observable,
-so APIs should treat them as if they no longer exist.
-They should not be visible to the "outside world" through document-spanning APIs
-(e.g. {{Clients/matchAll()|clients.matchAll()}}, {{Window/opener|window.opener}}).
+You should specify how it works with non-[=Document/fully active=] BFCached documents, following the guidelines in the
+[Supporting BFCached Documents](TODO) guide.
```
Comment (cynthia) [inline index.bs:705]:
```suggestion
[Supporting BFCached Documents](https://w3ctag.github.io/bfcache-guide/) guide.  
```

File: index.bs [hunk 2]
```diff
@@ -691,183 +691,23 @@ it is not always *sufficient*
...
-</div>
-<div class="example">
-  For network connections or streams,
-  the data received while not [=Document/fully active=] should be delivered only
-  when the document becomes [=Document/fully active=] again,
-  but whereas a stream might have created many events with a small amount of data each,
-  it could be delivered as smaller number of events with more data in each.
-</div>
+If your API does things that fall into **any** of the below categories:
+* Interacts with a document from the "outside" (e.g. sends information to a document)
```
Comment (cynthia) [inline index.bs:699]:
```suggestion
* Interacts with a document from the "outside"
  (e.g. sends information to a document)
```

File: index.bs [hunk 3]
```diff
@@ -691,183 +691,23 @@ it is not always *sufficient*
...
+* Makes cross-document interaction/resource sharing possible (e.g. holding locks)
+* May malfunction when a document is kept in a non-[=Document/fully active=] BFCached state (instead of getting destroyed)
+    after the user navigates away from it or gets restored (e.g. expects that a state saved in the document won’t span multiple navigations)
 
-<h4 id="omit-non-fully-active">Omit non-[=Document/fully active=] documents from APIs that span multiple documents</h4>
-Non-[=Document/fully active=] documents should not be observable,
-so APIs should treat them as if they no longer exist.
-They should not be visible to the "outside world" through document-spanning APIs
-(e.g. {{Clients/matchAll()|clients.matchAll()}}, {{Window/opener|window.opener}}).
+You should specify how it works with non-[=Document/fully active=] BFCached documents, following the guidelines in the
```
Comment (cynthia) [inline index.bs:704]:
```suggestion
Specify how your feature works
with non-[=Document/fully active=] BFCached documents,
following the guidelines in the
```

File: index.bs [hunk 4]
```diff
@@ -691,183 +691,23 @@ it is not always *sufficient*
...
-  [=Sticky activation=] is determined by the "last activation timestamp",
-  which is tied to a document.
-  This means after a user triggers activation once on a document,
-  the document will have sticky activation forever,
-  even after the user navigated away and back to it again.
-  Whether this should actually be reset when full activity is lost or not
-  is still [under discussion](https://github.com/whatwg/html/issues/6588).
-</div>
+Note: It is possible for a document to become non-[=Document/fully active=] for other reasons not related to BFcaching,
+such as when the iframe holding the document gets detached.
```
Comment (cynthia) [inline index.bs:708]:
```suggestion
such as when the iframe
holding the document gets detached.
```

File: index.bs [hunk 5]
```diff
@@ -691,183 +691,23 @@ it is not always *sufficient*
...
+This guideline only focuses on the case where BFCache is involved,
```
Comment (cynthia) [inline index.bs:709]:
```suggestion
This guidance only focuses
on the case where BFCache is involved,
```

File: index.bs [hunk 6]
```diff
@@ -691,183 +691,23 @@ it is not always *sufficient*
...
+and not other cases that might cause a document to become non-[=Document/fully active=].
```
Comment (cynthia) [inline index.bs:710]:
```suggestion
and not other cases that might
cause a document to become non-[=Document/fully active=].
```

File: index.bs (hunk 2 above)
Comment (cynthia) [reply index.bs:699]: Maybe "Interacts with an external document" could be better? Quotes tend to implicitly suggest that the quoted term is pseudo-something.

**Comment by cynthia** [comment]
Sorry this took so long, overall the proposed changes look fine to us - and we'd like to do a review of the BFCache guide seprately.

As for that, the repository is here: https://github.com/w3ctag/bfcache-guide
@rakina and @domenic should have write access (on top of the usual TAG suspects)

Wondering how we should review the bfcache-guide repository though, I suppose I could create a PR against the main branch so that we have something to review. Thoughts?

File: index.bs [hunk 7]
```diff
@@ -691,183 +691,26 @@ it is not always *sufficient*
...
-  Whether this should actually be reset when full activity is lost or not
-  is still [under discussion](https://github.com/whatwg/html/issues/6588).
-</div>
+<h3 id="support-non-fully-active">Support non-fully active BFCached documents</h3>
+
+Specify how your feature behaves
+in non-[=Document/fully active=] BFCached (Back/forward cached) documents if possible.
+
+If your feature does **anything**
+that fall into the
```
Comment (torgo) [inline index.bs:700]:
```suggestion
that falls into the
```

**Review by torgo — APPROVED** [review]
looks good to me.

**Review by cynthia — APPROVED** [review]

**Comment by rakina** [comment]
Thanks @cynthia and @torgo! Looks like https://github.com/w3ctag/bfcache-guide already has the doc committed so maybe we should move that into a separate branch and clear out the main branch, and then make a PR against the main branch?

**Pull request merged**